### ISSUE TITLE
Unmounting `ink-select-input` exits the process

### ISSUE THREAD

**Issue by SimenB** [vadimdemedes/ink#145]
```js
function PromptThing() {
  const [value, setValue] = React.useState('');

  const valueAsRegexp = new RegExp(value);
  const items = [
    { label: 'hah', value: 'thing number 1' },
    { label: 'hah2', value: 'thing number 2' },
  ].filter(item => valueAsRegexp.test(item.value));

  return (
    <Box flexDirection="column">
      <Text bold>Pattern Mode Usage</Text>
      <Box marginLeft={1} flexDirection="column">
        <Box marginY={1}>
          <TextInput value={value} onChange={setValue} />
        </Box>
        <Box>
          {value.length > 0 ? (
            <Box flexDirection="column">
              <Text>Pattern matches {items.length} files</Text>
              <SelectInput items={items} limit={10} />
            </Box>
          ) : (
            <Text>Start typing to filter by a filename regex pattern.</Text>
          )}
        </Box>
      </Box>
    </Box>
  );
}

render(<PromptThing />);
```

Enter a letter, then hit backspace. The process exits. Removing `SelectInput` from the component stops this behavior

```json
    "ink": "^2.0.3",
    "ink-select-input": "^3.0.1",
    "ink-text-input": "^3.0.0",
```

**Comment by vadimdemedes** [comment]
I think I know what it is, working on a fix!

**Comment by vadimdemedes** [comment]
Cause of this bug is Ink doesn't keep count how many components requested raw mode, so when first component calls `setRawMode(false)`, Ink disables raw mode, which makes the process naturally exit. Fixed via https://github.com/vadimdemedes/ink/commit/321aa1eac51a76eec40efcff1286f71d08cc5019, going to keep the issue open until new release is out. Thanks for reporting, @SimenB!

**Comment by vadimdemedes** [comment]
Released `2.0.4` with a fix.
